frontend/lib/resolution/Resolver.cpp

```diff
@@ -3079,13 +3079,16 @@ static bool computeTaskIntentInfo(Resolver& resolver, const NamedDecl* intent,
 }
 
 bool Resolver::enter(const ReduceIntent* reduce) {
-
   ID id;
   QualifiedType type;
   ResolvedExpression& result = byPostorder.byAst(reduce);
 
   if (computeTaskIntentInfo(*this, reduce, id, type)) {
     validateAndSetToId(result, reduce, id);
+    // set reduce intent shadow variable to a VAR with type of shadowed variable
```

Comment:
Wow the spec is kind of confusing here...

https://chapel-lang.org/docs/language/spec/task-parallelism-and-synchronization.html#task-intents

It looks like there is both a task variable and an outer variable in this instance?

So what you are setting as a var is the task variable, correct?

Reply:
My understanding was in the case of reduce intents, what we call the shadow variable is also basically a task variable, though not explicitly. So as a shadow, it has a corresponding outer variable, but it also essentially acts as a task-private variable with the added functionality of accumulation being handled.

Yes, I'm setting the task variable as a VAR. I decided that was appropriate based on these spec lines,
> At the start of each task the corresponding shadow variable is initialized to the identity value of the reduction operator. Within the task it behaves as a regular variable.

as well as the [Reduce Intents tech note](https://chapel-lang.org/docs/technotes/reduceIntents.html#readme-reduceintents) which echoed this in more detail. I felt a little uneasy with this decision at first but I don't think there's another option that makes more sense, since the variable is essentially a task-private variable here.